src/localPRG.cpp

```diff
@@ -1183,62 +1184,110 @@ LocalPRG::find_alt_path(const std::vector<LocalNodePtr> &ref_path, const uint32_
+    uint32_t number_of_bases_in_local_path_before_the_position = 0;
+    for (const auto &local_node : local_path) {
+        bool local_node_is_empty = local_node->pos.length == 0;
+        if (local_node_is_empty) {
             continue;
-        } else if (n->pos.get_end() < kmer_path[1]->path.get_start()) {
-            added += n->pos.length;
-        } else if (n->pos.get_end() >= kmer_path[1]->path.get_start() and n->pos.start < kmer_path[1]->path.get_end()) {
-            added += kmer_path[1]->path.get_start() - n->pos.start;
+        }
+
+        bool local_node_starts_before_position = local_node->pos.start < position;
+        bool local_node_ends_before_position =
+                local_node->pos.get_end() <= position;
+        if (local_node_ends_before_position) {
+            number_of_bases_in_local_path_before_the_position += local_node->pos.length;
+        } else if (local_node_starts_before_position) {
+            number_of_bases_in_local_path_before_the_position += position - local_node->pos.start;
             break;
         }
     }
 
-    //std::cout << "added " << added << std::endl;
-    KmerNodePtr prev = nullptr;
-    for (const auto &n: kmer_path) {
-        //std::cout << n->path << " ";
-        if (n->path.length() == 0)
+    return number_of_bases_in_local_path_before_the_position;
+}
+
+uint32_t
+LocalPRG::get_number_of_bases_that_are_exclusively_in_the_previous_kmer_node(const KmerNodePtr &previous_kmer_node,
+                                                                             const KmerNodePtr &current_kmer_node) const {
+    uint32_t number_of_bases_that_are_exclusively_in_the_previous_kmer_node = 0;
+    auto previous_kmer_node_path_interval_iterator = previous_kmer_node->path.begin();
+
+    bool previous_kmer_node_path_interval_ends_before_current_kmer_node =
+            previous_kmer_node_path_interval_iterator->get_end() <=
+            current_kmer_node->path.get_start();
+    while (previous_kmer_node_path_interval_ends_before_current_kmer_node) {
+        number_of_bases_that_are_exclusively_in_the_previous_kmer_node += previous_kmer_node_path_interval_iterator->length;
+        previous_kmer_node_path_interval_iterator++;
+        previous_kmer_node_path_interval_ends_before_current_kmer_node =
+                previous_kmer_node_path_interval_iterator->get_end() <=
+                current_kmer_node->path.get_start();
+    }
+
+    if (current_kmer_node->path.get_start() > previous_kmer_node_path_interval_iterator->start) {
+        number_of_bases_that_are_exclusively_in_the_previous_kmer_node +=
+                current_kmer_node->path.get_start() - previous_kmer_node_path_interval_iterator->start;
+    }
+
+    return number_of_bases_that_are_exclusively_in_the_previous_kmer_node;
+}
+
+
+std::pair<std::vector<uint32_t>, std::vector<uint32_t>>
+LocalPRG::append_kmer_covgs_in_range(const KmerGraphWithCoverage &kmer_graph_with_coverage,
```

Comment:
The name of this function is misleading now. It says append kmer coverages, but it *returns* coverages. Better name might be `forward_and_reverse_kmer_covgs_in_range`

Reply:
Oh, that is actually an important catch! Refactoring it